### PR TITLE
LLT-4661: update wireguard-go and related dependencies

### PR DESCRIPTION
**Pull request by tomaszklak**
### Problem
We are using 2 years old wireguard-go version and some other outdated Go dependencies

### Solution
All the Go dependencies are updated to the latest stable versions. All of telio specific code including `OpenOnLocalhost` is moved to new files (`bind_telio_std.go` and `bind_telio_windows.go`). Other files are mostly copied from the upstream and modified slightly to compile as part of the `main` package - to view the real differences you can diff against upstream, eg:

```
$ curl 'https://git.zx2c4.com/wireguard-go/plain/conn/bind_windows.go?id=12269c2761734b15625017d8565745096325392f' | diff bind_windows.go -
```

### :ballot_box_with_check: Definition of Done checklist
- [x] Commit history is clean ([requirements](../blob/main/docs/git_commit_messages_requirements.md))
- [x] README.md is updated
- [x] changelog.md is updated
- [x] Functionality is covered by unit or integration tests

### CODE REVIEW

File: wireguard-go-rust-wrapper/wireguard-go/controlfns_unix.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
+//go:build !windows && !linux && !wasm
```
Comment (mathiaspeters) [inline wireguard-go-rust-wrapper/wireguard-go/controlfns_unix.go:1]: We currently only support wg-go on windows. Is this file required anyways, or was there an otherwise active decision to keep files for unix compat?

Comment (mathiaspeters) [reply wireguard-go-rust-wrapper/wireguard-go/controlfns_unix.go:1]: Same question for `controlfns_linux.go`

File: wireguard-go-rust-wrapper/wireguard-go/build.sh [hunk 2]
```diff
@@ -2,10 +2,10 @@
 
 set -eux
 
-if [[ $1 != windows ]]; then
```
[the remark this answers is not in the document]
Comment (tomaszklak) [reply wireguard-go-rust-wrapper/wireguard-go/build.sh:5]: Yeah, I missed that I haven't removed it. Will do that in a sec

File: wireguard-go-rust-wrapper/wireguard-go/controlfns_unix.go (hunk 1 above)
Comment (tomaszklak) [reply wireguard-go-rust-wrapper/wireguard-go/controlfns_unix.go:1]: I've decided to copy all versions of controlfns_* as is to make it simpler when comparing to the upstream (and also when building locally on different OSes)

**Comment by mathiaspeters** [comment]
+0.5

**Review by gytsto — COMMENTED** [review]
+0.5

**Review by jjanowsk — APPROVED** [review]
+1
